src/libraries/System.Reflection.Emit/src/System/Reflection/Emit/SignatureHelper.cs

```diff
@@ -0,0 +1,86 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Reflection.Metadata;
+using System.Reflection.Metadata.Ecma335;
+
+namespace System.Reflection.Emit
+{
+    // TODO: Only support simple signatures. More complex signatures will be added.
+    internal static class MetadataSignatureHelper
+    {
+        internal static BlobBuilder FieldSignatureEncoder(Type fieldType)
+        {
+            var fieldSignature = new BlobBuilder();
+
+            WriteSignatureTypeForReflectionType(new BlobEncoder(fieldSignature).FieldSignature(), fieldType);
+
+            return fieldSignature;
+        }
+
+        internal static BlobBuilder MethodSignatureEncoder(Type[]? parameters, Type? returnType, bool isInstance)
+        {
+            // Encoding return type and parameters.
+            var methodSignature = new BlobBuilder();
+
+            ParametersEncoder parEncoder;
+            ReturnTypeEncoder retEncoder;
+
+            new BlobEncoder(methodSignature).
+                MethodSignature(isInstanceMethod: isInstance).
+                Parameters((parameters == null) ? 0 : parameters.Length, out retEncoder, out parEncoder);
+
+            if (returnType != null && returnType.FullName != "System.Void")
```

Comment:
Similarly, these matches should be done by comparing with the (cached) `System.Void` Type instance.